File: 프로그래머스/unrated/120837. 개미 군단/개미 군단.js
function solution(hp) {
    
    
    const generalAnt = Math.floor(hp / 5)
    const soldierAnt = Math.floor((hp - (generalAnt * 5)) / 3)
    const workerAnt = hp - ((generalAnt * 5) + (soldierAnt * 3))
    return generalAnt + soldierAnt + workerAnt
}

// function solution(hp) {
    
//     var count = 0;
//     while(hp >= 5) {
//         hp -= 5
//         count++
//     }
    
//     while(hp >= 3) {
//         hp -= 3
//         count++
//     }
    
//     while(hp > 0) {
//         hp -= 1
//         count++
//     }
//     return count
// }

File: 프로그래머스/0/120837. 개미 군단/개미 군단.kt
class Solution {
    fun solution(hp: Int): Int {
        val generalAnt = hp / 5  
        val remainAfterGeneral = hp % 5  
        val soldierAnt = remainAfterGeneral / 3  
        val workerAnt = remainAfterGeneral % 3 
        
        return generalAnt + soldierAnt + workerAnt
       
    }
}